designdocs/provider_contexts.md

```diff
@@ -0,0 +1,353 @@
+
+### **Storage Migration** <a id='migration'></a>
+
+To support the new structure, a storage migration will be required to:
+
+- Migrate existing `ProviderToRegistryEntry` to the new format with default values for localized names and logos.
+
+### **Mainnet Approval Flow** <a id='governance'></a>
+
+_Any_ change to the Provider or Application Context must be approved by governance.
+For now, that governance approval is any single Frequency Council member.
+
+```mermaid
+sequenceDiagram
+    participant P as Provider
+    participant D as Provider Dashboard
+    participant F as Frequency
+    participant G as Frequency Council
+
+    note over P,G: Provider Registration
+    P->>D: Login
+    D-->>+F: Create MSA
+    F-->>-D: MSA created
+    P->>D: Request to be Provider form
+    P->>D: Add Applications as needed
+    D-->>+F: Generate Frequency Council Proposal
+    G->>F: Review and Approve Proposal
+    F->>F: Execute Provider registration
+    F-->>-D: MSA is a Provider
+
+    note over P,G: Provider Update Registration
+    P->>D: Login
+    P->>D: See Provider and Application Information
+    P->>D: Request to update Provider form
+    P->>D: Update/add Applications as needed
+    D-->>+F: Generate Frequency Council Proposal
+    G->>F: Review and Approve Update Proposal
+    F->>F: Execute Provider changes as an upsert
+    F-->>-D: Update display
+    P->>D: See Updated Provider and Application Information
+```
+
+#### Open Questions
+
+- Is this a set or edit pattern for the applications?
+- Should adding/updating applications be a separate call?
+- Should adding/updating a translation be a separate call?
+
+#### Implementation Suggestion
+
+To limit the amount of unapproved bytes intended to be interpreted as images on chain, an option would be to merely submit the hashes of images for approval.
+This hash would then be allowlisted for that provider.
+Once approved, that provider could submit the image to chain.
+In the governance process, a link to the image would need to be submitted.
+
+### **Example of Wallet Usage via SIWF** <a id='siwf'></a>
+
+SIWF [Signed Request Payload](https://projectlibertylabs.github.io/siwf/v2/docs/DataStructures/All.html) can expand the optional `applicationContext` value with a new optional `id` field that is the text of the `ApplicationIndex` on chain.
+This would allow a smooth transition between `applicationContext.url` and `applicationContext.id` with both being optional.
+
+The Wallet would then:
+
+1. Verify the SIWF Signed Request.
+2. Lookup the Provider via the `publicKey` in the SIWF Signed Request.
+3. If any, fetch the `ApplicationIndex` from Frequency.
+4. If present, fetch the `ProviderToApplicationRegistryEntry` for that (ProviderId, ApplicationId) pair. Otherwise, fetch the default `ProviderToRegistryEntry` for the ProviderId
```

Comment:
Do we just keep provider side registry to contain just the name and empty for others and instead create a `default` application with id:0 when a provider is registered?

 that way there is atleast one application is   for providers and mutations can be limited to applications root and keep `ProviderToRegistry` immutable

Reply:
I do like the idea of an "application 0" being the default. `ProviderToRegistry` could be more minimal. hmm... I'm open to that solution. Aka all Providers have at least one application. Makes the data storage simpler for sure.

Reply:
using a hash table `ApprovedLogos` makes it easy now, since `ProviderRegitry` would not contain raw bytes of logos but hashes and a second call to logo tables can resolve hash to image for users, so I believe we dont need a application 0 if fallback is `ProviderRegistry`